### PR TITLE
Update IvyError and add RequestHandler class

### PR DESCRIPTION
**Pull request by null68**

### CODE REVIEW

File: packages/ivycord.js/src/errors/IvyError.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+class IvyError extends Error {
+  constructor(error: Errors, message?: string) {
+    const errorMessage = messages[error].replace(/{message}/g, message ?? '');
+    super(errorMessage);
+  }
+}
+
+export { IvyError };
```
Comment (jvvan) [inline packages/ivycord.js/src/errors/IvyError.ts]:
```suggestion
const messages = {
  WS_ALREADY_CONNECTED: 'Already connected to the WebSocket',
  LARGE_THRESHOLD_INVALID: 'Large threshold must be between 50 and 250',
  FETCH_ERROR: 'An error occurred while fetching\n\n{message}'
};

type ErrorsKey = keyof typeof messages;

class IvyError extends Error {
  constructor(error: ErrorsKey, message?: string) {
    const errorMessage = messages[error].replace(/{message}/g, message ?? '');
    super(errorMessage);
  }
}

export { IvyError };
```

File: packages/ivycord.js/src/handlers/RequestHandler.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(client: BaseClient) {
+    this.client = client;
+  }
+  async request(method: Method, endpoint: string, body: unknown) {
+    try {
+      const headers = {
+        'Content-Type': 'application/json',
+        'User-Agent': `DiscordBot (Ivycord/${
+          // eslint-disable-next-line @typescript-eslint/no-var-requires
+          require('../../package.json').version
```
Comment (jvvan) [inline packages/ivycord.js/src/handlers/RequestHandler.ts]: move to a constants file?

File: tsconfig.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
     // "resolvePackageJsonExports": true,                /* Use the package.json 'exports' field when resolving package imports. */
     // "resolvePackageJsonImports": true,                /* Use the package.json 'imports' field when resolving imports. */
     // "customConditions": [],                           /* Conditions to set in addition to the resolver-specific defaults when resolving imports. */
-    // "resolveJsonModule": true,                        /* Enable importing .json files. */
+    "resolveJsonModule": true /* Enable importing .json files. */,
```
Comment (jvvan) [inline tsconfig.json:42]: put the comma before the comment

File: packages/ivycord.js/src/handlers/RequestHandler.ts (hunk 2 above)
Comment (kikorp78) [reply packages/ivycord.js/src/handlers/RequestHandler.ts]: Done.

Comment (kikorp78) [reply packages/ivycord.js/src/handlers/RequestHandler.ts]: Done.

File: tsconfig.json (hunk 3 above)
Comment (kikorp78) [reply tsconfig.json:42]: Prettier prevents this.

**Review by kikorp78 — APPROVED** [review]

**Review by camdzic — APPROVED** [review]

**Pull request merged**